File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/AndonManageAddCommand.cs
using SIE.Andon.Andons;
using SIE.Domain;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.Andon.Andons.Commands
{
    /// <summary>
    /// 安灯管理触发命令
    /// </summary>
    public class AndonManageAddCommand : ViewCommand
    {
        /// <summary>
        /// 触发
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        /// <exception cref="NotImplementedException"></exception>
        protected override object Excute(ViewArgs args, string scope)
        {
            var andonManage = args.Data.ToJsonObject<AndonManage>();
            var dateTimeNow = RF.Find<AndonManage>().GetDbTime();
            andonManage.AndonManageCode = RT.Service.Resolve<AndonManageController>().GetAndonManageCode();
            andonManage.State = SIE.Andon.Andons.Enum.AndonManageState.Standby;
            andonManage.TriggerId = RT.IdentityId;
            andonManage.TriggerTime = dateTimeNow;
            andonManage.FaultTime = dateTimeNow;
            andonManage.WorkGroup = RT.Service.Resolve<AndonManageController>().GetLoaderWorkGroup();
            return andonManage;
        }
    }
}

File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/AndonManageAddCommand.js
SIE.defineCommand('SIE.Web.Andon.Andons.Commands.AndonManageAddCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "触发", group: "edit", iconCls: "icon-AddEntity icon-green" },
    showView: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: model,
                success: function (res) {
                    var code = res.Result.AndonManageCode;
                    var triggerId = res.Result.TriggerId;
                    var triggerTime = res.Result.TriggerTime;
                    var faultTime = res.Result.FaultTime;
                    var workGroup = res.Result.WorkGroup;
                    var state = res.Result.State
                    CRT.Workbench.addPage({
                        entityType: me.view.model,
                        recordId: entity.getId(),
                        title: me.getEditViewTitle(entity),
                        params: {
                            AndonManageCode: code,
                            TriggerId: triggerId,
                            TriggerTime: triggerTime,
                            FaultTime: faultTime,
                            WorkGroup: workGroup,
                            State:state,
                        },
                        isDetail: true
                    });
                }
            }, me.view);
        }
    },
});
